### PR TITLE
feat(npm dependencies): add typings support for custom npm dependencies

### PR DESCRIPTION
**Pull request by nemni8**

### CODE REVIEW

File: test/utils/typesByContext.js [hunk 1]
```diff
@@ -45,13 +53,32 @@ const createDeclarationsFilesByConfigPath = (
...
 };
 
 const getDynamicPageTypings = testTmpDirPath => ({
   elementsMap: handlePageElementsMap(testTmpDirPath),
-  widgets: handleWidget(testTmpDirPath)
+  widgets: handleWidget(testTmpDirPath),
+  dependencies: handleDependencies(testTmpDirPath)
 });
 
+const handleDependencies = (testPath, shouldWrite = false) => {
```
Comment (dor-itzhaki) [inline test/utils/typesByContext.js:65]: these functions (all the handle functions) do 2 things:
1. read the test example path, according to some structure we've defined for the tests
2. use corvid-types API to and save the result to some new file

so for example, the test will look like this:

const context = examplesParser.parse(testPath)
const declarations = corvidTypes.declarations(context)
expect(declarations)......



i think that:
#1 should be part of test utils (stay here)
#2 should be moved to the test itself

File: src/index.js [hunk 2]
```diff
@@ -28,31 +33,50 @@ const getDynamicTypes = ({ elementsMap, widgets }) => {
...
+
 module.exports = {
   configPaths: {
     page: `corvid-types/${TS_CONFIG_PATHS.PAGES}`,
     backend: `corvid-types/${TS_CONFIG_PATHS.BACKEND}`,
     public: `corvid-types/${TS_CONFIG_PATHS.PUBLIC}`
   },
   declarations: {
-    page: ({ elementsMap, widgets } = {}) => {
+    page: ({ dependencies, elementsMap, widgets } = {}) => {
```
Comment (dor-itzhaki) [inline src/index.js:52]: maybe call it npmDependencies (not sure) ?

File: src/index.js [hunk 3]
```diff
@@ -28,31 +33,50 @@ const getDynamicTypes = ({ elementsMap, widgets }) => {
...
+        ...fullCorvidTypes.PUBLIC,
+        ...getNpmDependenciesTypesDeclarations(dependencies)
+      ];
     }
   },
   getWixModulesList: () => wixModulesNames,
   getWidgetTypeDeclarations,
-  getPageElementsTypeDeclarations
+  getPageElementsTypeDeclarations,
+  getNpmDependenciesTypesDeclarations
```
[the remark this answers is not in the document]
Comment (nemni8) [reply src/index.js:81]: not only for tests, in the local mode we use the `getPageElementsTypeDeclarations` in order to create the pageElementsMap typing. so those function will be probably be called from LM when widgets and npm dependencies will be supported

File: test/it/code-samples/positive/backend-custom-npm-dependencies/dependencies.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
[the remark this answers is not in the document]
Comment (nemni8) [reply test/it/code-samples/positive/backend-custom-npm-dependencies/dependencies.json:1]: true, but these tests are checking general logic of npm dependencies with no relation to wix implementation

File: src/index.js (hunk 2 above)
Comment (nemni8) [reply src/index.js:52]: I think I will stick with dependencies because it is a more generic name

File: test/utils/typesByContext.js (hunk 1 above)
Comment (nemni8) [reply test/utils/typesByContext.js:11]: I did some refactoring in order to make the test setup flow more readable

**Pull request merged**